src/animation/interpolate.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2022 The Druid Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use druid::{Color, Insets, Point, Rect, Size, Vec2};
+
+/// Interpolate between two values
+///
+/// Interpolate between `self` and `other` where `value` is the
+/// position (between 0 and 1). For example, a simple linear
+/// interpolation is implemented as: `self + (other - self) * value`
+pub trait Interpolate: PartialEq + Clone {
```

Comment:
Could possibly be better to call this `Lerp` to distinguish between this and other non-linear interpolations.